weldx/transformations/local_cs.py

```diff
@@ -292,35 +282,25 @@ def _build_orientation(
         return orientation
 
     @classmethod
-    def _build_coordinates(cls, coordinates, time: pd.DatetimeIndex = None):
-        """Create xarray coordinates from different formats and time-inputs.
-
-        Parameters
-        ----------
-        coordinates:
-            Coordinates data.
-        time:
-            Valid time index formatted with `_build_time_index`.
-
-        Returns
-        -------
-        xarray.DataArray
-
-        """
+    def _build_coordinates(
+        cls, coordinates, time: Union[Time, None]
+    ) -> Union[xr.DataArray, TimeSeries]:
+        """Create xarray coordinates from different formats and time-inputs."""
         if isinstance(coordinates, TimeSeries):
             if coordinates.is_expression:
                 return coordinates
             coordinates = cls._coords_from_discrete_time_series(coordinates)
 
         if coordinates is None:
-            coordinates = np.array([0, 0, 0])
+            coordinates = np.zeros(3)
 
         if not isinstance(coordinates, xr.DataArray):
-            time_coordinates = None
             if not isinstance(coordinates, (np.ndarray, pint.Quantity)):
                 coordinates = np.array(coordinates)
-            if coordinates.ndim == 2:
-                time_coordinates = time
+
+            time_coordinates = (
+                time.as_timedelta_index() if coordinates.ndim == 2 else None
+            )
```

Comment:
I think we should simply add support for passing `Time` in `xr_3d_vector` and `xr_3d_matrix` instead of handling this here

Reply:
> I think we should simply add support for passing `Time` in `xr_3d_vector` and `xr_3d_matrix` instead of handling this here

Changed it as suggested. Have a look into both of the updated functions